embassy-boot/src/firmware_updater/asynch.rs

```diff
@@ -210,6 +258,7 @@ impl<'d, DFU: NorFlash, STATE: NorFlash> FirmwareUpdater<'d, DFU, STATE> {
 pub struct FirmwareState<'d, STATE> {
     state: STATE,
     aligned: &'d mut [u8],
+    last_erased_dfu_page_index: Option<usize>,
```

Comment:
I think it makes more sense to keep this inside the FirmwareUpdater type, because it's only used by that, and the FirmwareState can be used standalone where it will not be using this field. (The 'state' in FirmwareState refers to it managing the state partition, not being the state of the firmware updater).

Reply:
Agreed. Done in [56e6b6b](https://github.com/embassy-rs/embassy/pull/2543/commits/56e6b6bee6bd6087b35857e8aa1a011f6e83f703).

Thanks for the review @lulf!